### PR TITLE
Add Prometheus Receiver tests to run OpenMetrics test cases

### PR DESCRIPTION
**Pull request by mustafain117**
**Description:** <Describe what has changed.>
In order to verify that the Prometheus Receiver is functioning as expected for transforming OpenMetrics exposition to OTLP format, this PR adds two tests to run positive and negative test cases form the OpenMetrics repository (https://github.com/OpenObservability/OpenMetrics/tree/main/tests/testdata/parsers).

`TestOpenMetricsPositive` test:

- Runs positive OpenMetrics test cases, asserts scrape is successful

`TestOpenMetricsNegative` test:

- Runs negative OpenMetrics test cases, asserts scrape is unsuccessful

**Link to tracking Issue:** <Issue number if applicable>
Issue 6106

### CODE REVIEW

File: receiver/prometheusreceiver/metrics_receiver_open_metrics_test.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+		}
+		targets = append(targets, testData)
+	}
+
+	testComponent(t, targets, false, "", true)
+}
+
+// maps each test name to the test data from OpenMetrics repository
+func getOpenMetricsTestData(negativeTestsOnly bool) map[string]string {
+	response, err := http.Get(testDataURL)
```
Comment (Aneurysm9) [inline receiver/prometheusreceiver/metrics_receiver_open_metrics_test.go:87]: I'm not sure I really like the idea of making tests depend on internet connectivity and github availability.  How big are the test files? Can we keep a copy of them here in a `testdata` directory?

Comment (mustafain117) [reply receiver/prometheusreceiver/metrics_receiver_open_metrics_test.go:87]: The size of the test data folder from [OpenMetrics](https://github.com/OpenObservability/OpenMetrics/tree/main/tests/testdata/parsers) is 1.7MB, removing the `test.json` (test metadata) files which are unused reduces the size to 938 KB.

Comment (mustafain117) [reply receiver/prometheusreceiver/metrics_receiver_open_metrics_test.go]: @Aneurysm9 Removed HTTP requests and dependency on github availability by copying OpenMetrics test data into the `testdata` directory.

**Review by Aneurysm9 — APPROVED** [review]

**Comment by mustafain117** [comment]
Merged upstream.

**Pull request closed without merging**